addons/web_qweb/static/js/qweb_view.js

```diff
@@ -0,0 +1,139 @@
+odoo.define('web_qweb', function (require) {
+"use strict";
+
+var core = require('web.core');
+
+var AbstractView = require('web.AbstractView');
+var AbstractModel = require('web.AbstractModel');
+var AbstractRenderer = require('web.AbstractRenderer');
+var AbstractController = require('web.AbstractController');
+
+var _lt = core._lt;
+var registry = require('web.view_registry');
+
+var Model = AbstractModel.extend({
+    init: function () {
+        this._super.apply(this, arguments);
+        this._body = null;
+        this._model_name = null;
+        this._view_id = false;
+    },
+    _fetch: function (domain, context) {
+        var _this = this;
+        return this._rpc({
+            model: this._model_name,
+            method: 'qweb_render_view',
+            kwargs: {
+                view_id: this._view_id,
+                domain: domain,
+                context: context
+            }
+        }).then(function (r) {
+            _this._body = r;
+            return r;
+        });
+    },
+    get: function () {
+        return this._body;
+    },
+    load: function (params) {
+        this._view_id = params.view_id;
+        this._model_name = params.modelName;
+
+        return this._fetch(
+            params.domain,
+            params.context
+        );
+    },
+    reload: function (_id, params) {
+        return this._fetch(
+            params.domain,
+            params.context
+        )
+    }
+});
+var Renderer = AbstractRenderer.extend({
+    _render: function () {
+        var _this = this;
```

Comment:
var self = this;

Reply:
@ged-odoo I've seen this all around in the framework and I'm curious. What is the reason that you do that in every function within JS pretty much? Why not work with this instead of self? I think I miss something important and I'd like to know why.
var self = this;

Reply:
@Yenthe666 javascript's historical functions (`function () {}` as opposed to the "new" ES6 arrow functions) resolve variables lexically (checking the next scope if it's not local)… except for `this` (and possibly `arguments`, that one's a bit weird). Instead it's either:

* provided explicitly on call/bind (`fn.call(explicit_this, …params)` or `fn.bind(explicit_this)`)
* provided implicitly through the `.` operator (e.g. `config.fn()` <- the `config` object will be set as `this`)
* the [global scope object](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope) in sloppy mode
* `undefined` in strict mode

So if one wants to use the current this in a sub-function (closure), it has to be bound to a regular local variable (as those are resolved lexically).

Reply:
@odony and @xmo-odoo a really big thank you for explaining this - no pun 😉 - and adding links to relevant topics, I've learned a valuable lesson today. Keep up the good work, really appreciate it!